subsys/bluetooth/services/fast_pair/fp_gatt_service.c

```diff
@@ -61,7 +66,8 @@ struct msg_key_based_pairing_req {
 	uint8_t msg_type;
 	uint8_t fp_flags;
 	uint8_t provider_address[BT_ADDR_SIZE];
-	uint8_t seeker_address[BT_ADDR_SIZE];
+	uint8_t remaining_msg[FP_CRYPTO_AES128_BLOCK_LEN - sizeof(uint8_t) - sizeof(uint8_t) -
```

Comment:
The seeker address field was wrong anyway. It is optional field so I would prefer to throw it out of this structure. I propose to have `msg_key_based_pairing_write` structure to hold both key-based pairing request and action request for validation purpose and `msg_action_req` structure to hold action request. `msg_action_req` strucutre would be parsed from `msg_key_based_pairing_write` as it's detailed version of the same message.

Reply:
I would prefer to use `seeker_address` instead of `remaining_msg`. That name seems cleaner.
Our code is aware that the address may not be set in some messages (then we also do not use it in our implementation).

Reply:
If we use structures as an abstraction for messages, then there is no need to pass salt in the structure too (salt should be handled at our transport level, it's not needed later on).